### PR TITLE
[ML] Close any opened pipes if there is an error connecting to the process

### PR DESCRIPTION
**Pull request by davidkyle**
In the native process factories there is a point where the input and output streams can leak without being closed if there is an error connecting to the process.

https://github.com/elastic/elasticsearch/blob/76ad7d84646a65f2ceb12ffada68ca3126a97255/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/normalizer/NativeNormalizerProcessFactory.java#L77

There is a bit of an encapsulation failure here as once `ProcessPipes::connectStreams` has completed those new streams are passed to the native process abstraction class then neatly closed when the native process is closed. Problem is `ProcessPipes::connectStreams` creates multiple streams and if the Nth connection fails the earlier successful ones should be cleared up.; admittedly `ProcessPipes::connectStreams` is most likely to error on the first connection but you never know.

### CODE REVIEW

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/process/NativeAnalyticsProcessFactory.java [hunk 1]
```diff
@@ -88,6 +88,11 @@ private void createNativeProcess(String jobId, AnalyticsProcessConfig analyticsP
         } catch (IOException e) {
             String msg = "Failed to launch data frame analytics process for job " + jobId;
             LOGGER.error(msg);
+            try {
```
Comment (dimitris-athanasiou) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/process/NativeAnalyticsProcessFactory.java]: I wonder if we could catch the exception in `ProcessPipes.connectStreams`, close unused streams and rethrow. This way callers would not need to worry about this.

Comment (davidkyle) [reply x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/process/NativeAnalyticsProcessFactory.java]: Well that seems obvious now. I made the change

**Review by dimitris-athanasiou — APPROVED** [review]
LGTM

**Comment by davidkyle** [comment]
run elasticsearch-ci/packaging-sample

**Pull request merged**